### DISCUSSION TITLE
What is your favorite maven project that is NOT working with m2eclipse?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-m2e/m2e-core#1064] [General]
From time to time there are some complains that m2eclipse is "not working well" or "it works better with X" and alike.

As it is hard to improve on vague complains, I'd like to start this discussion to collect some "facts", e.g. I'd like to ask what is your favorite (open source) maven project you like to use with m2eclipe but having issues. The goal is to use such project as a blue-print to check how m2e can be improved to make working with that project more smooth!

To suggest a project, please fill the following "checklist:

1. Give a link to the **public source repository** where the project is located
2. Give the **maven command** one usually uses to build the project
3. Describe what **issues hold you back using m2eclipse** with the project (e.g. it is to slow, fails to compile, missing support for, ...)
4. Optionally **describe alternative ways** you currently use (e.g. build on the command line, other IDE or plugin, ...)

Please try out the latest snapshot build before you submit the proposal, you can use this p2 repository:
https://download.eclipse.org/technology/m2e/snapshots/latest/

that way we make sure you are using the latest bugfixes and improvements!

**Comment by laeubi** [comment]
I'll start here with a project I recently used to improve some things in m2e:

1. Repository: https://github.com/apache/camel
2. `mvn clean install -T1C -DskipTests`
3. It takes very long for m2e to import the project and also very long to build after changes. Also there are a lot of mojo steps not covered by the execution
4. I usually use the commandline for this to build and only import dedicated projects

For the "camel-case" m2eclipse was recently already improved, import is now much faster but builds are still slow and one gets a huge list of not covered phases. To be fair, building the project on the commandline is also heavy with 16CPU + SSD it takes maven about **15 Minutes** to build!

**Reply by jochenw** [reply]
Lately, I notice, that some plugins, that are related to M2E, cannot be installed with the latest M2E. For example:

  - I can install M2E, and I can install Subversive, but I cannot install the "Subversive SVN Integration for the M2E project". (Last tested with Eclipse 4.26, and Subversive 4.8.0.v20220904-1901.
  - I can install M2E, and I can install the Groovy Eclipse plugin from the Marketplace, but only without the component "Groovy-Eclipse M2E integration 4.8.0.v202212311456-e2212-RELEASE"

I do not know, whether the problem lies with the other components, or with M2E. Given the fact, that there are several such incompatibilities, I am guessing M2E.

**Reply by laeubi** [reply]
Most likley these components are not compatible with m2e 2.x.

There are some options:

- Ask the project to update to m2e 2.x.
- Check if they are actually required (e.g. what feature are missing without them)
- Check if similar things can be archives by adding m2e support directly to the maven plugins or m2e
